src/spaces/addSpace.js

```diff
@@ -79,13 +79,33 @@ export async function loadAddSpaces() {
   let cardMap = {};
 
   for (let spaceData of spaces) {
-    let card = cardTemplate.content.cloneNode(true);
-    card.querySelector("img").src = browser.runtime.getURL(`/recipes/${spaceData.recipeId}/icon.svg`);
-    card.querySelector(".name").textContent = spaceData.title;
-    card.querySelector(".card")._spaceData = spaceData;
-
-    cardMap[spaceData.recipeId] = card.firstElementChild;
-    allCardsContainer.appendChild(card);
+    // Clone template content, then get the real element from the fragment
+    const frag = cardTemplate.content.cloneNode(true);
+    const cardEl = frag.firstElementChild;
+
+    // Populate visuals
+    cardEl.querySelector("img").src = browser.runtime.getURL(`/recipes/${spaceData.recipeId}/icon.svg`);
+    cardEl.querySelector(".name").textContent = spaceData.title;
+
+    // A11y and keyboard activation
+    cardEl.setAttribute("tabindex", "0");
+    cardEl.setAttribute("role", "button");
+    cardEl.setAttribute("aria-label", spaceData.title);
+    cardEl.addEventListener("keydown", (e) => {
+      if (e.key === "Enter" || e.key === " " || e.key === "Spacebar") {
+        e.preventDefault();
+        cardEl.click();
+      }
+    });
+
+    // Store data on the element
```

Comment:
I think we can remove most of the inline comments added. I tend to make comments when it is really hard to understand what is happening, but most of this appears straightforward.